### PR TITLE
clang_parser: return with error code if clang_parser finds an error

### PR DESCRIPTION
**Pull request by mmarchini**
Fixes: https://github.com/iovisor/bpftrace/issues/653

### CODE REVIEW

File: src/clang_parser.cpp [hunk 1]
```diff
@@ -230,8 +230,25 @@ void ClangParser::parse(ast::Program *program, BPFtrace &bpftrace, std::vector<s
       &translation_unit);
   if (error)
   {
-    std::cerr << "Clang error while parsing C definitions: " << error << std::endl;
-    std::cerr << "Input (" << input.size() << "): " << input << std::endl;
+    if (bt_debug == DebugLevel::kFullDebug) {
```
Comment (ajor) [inline src/clang_parser.cpp]: Won't this mean we just have silent errors unless running in debug mode?

File: src/clang_parser.cpp [hunk 2]
```diff
@@ -230,8 +230,25 @@ void ClangParser::parse(ast::Program *program, BPFtrace &bpftrace, std::vector<s
...
+      std::cerr << "Clang error while parsing C definitions: " << error << std::endl;
+      std::cerr << "Input (" << input.size() << "): " << input << std::endl;
+    }
+    clang_disposeTranslationUnit(translation_unit);
```
Comment (ajor) [inline src/clang_parser.cpp]: We shouldn't have multiple exit points where we have to include these clean up lines - they are the kind of thing that could get forgotten in the future

Comment (mmarchini) [reply src/clang_parser.cpp]: I tried to come up with a solution to that with minimal code changes but had no success. Let me give it another shot.

File: src/clang_parser.cpp (hunk 1 above)
Comment (mmarchini) [reply src/clang_parser.cpp]: We still get warnings and errors from clang, which is more useful/actionable for the user than this information.

Comment (mmarchini) [reply src/clang_parser.cpp]: Also, I don't think I ever hit this branch, even if the clang parsing fails (which is why we check for clang diagnostics below).

File: src/clang_parser.cpp (hunk 2 above)
Comment (mmarchini) [reply src/clang_parser.cpp]: Done, the second commit addresses this issue. Let me know if that approach works for you. If it does, I'll squash both commits.

File: src/clang_parser.cpp (hunk 1 above)
Comment (ajor) [reply src/clang_parser.cpp]: If I'm remembering right, this is the error we see in the LLVM5 static builds. If you don't see it in any other case we might as well keep it in non-debug mode

Comment (ajor) [reply src/clang_parser.cpp]: oh skipped your first comment here - yeah guess it's fine to keep this for debug mode then

File: src/clang_parser.cpp (hunk 2 above)
Comment (ajor) [reply src/clang_parser.cpp]: Thanks, I'll read it over tomorrow

**Comment by ajor** [comment]
Tests need updating

**Pull request merged**